go.mod

```diff
@@ -1,5 +1,10 @@
 module github.com/josharian/impl
 
-go 1.14
+go 1.19
```

Comment:
Can we make this 1.18 instead? That's the first generics release.

Reply:
So I was originally going to ask about this, because my understanding of the `go` directive in `go.mod` was that it signaled the _latest_ version of Go that a module had been tested with; I had a reason for this belief that I'm not braining properly enough to articulate at the moment, something something backwards compatibility vs forwards compatibility.

But as I was seeking out the source for that understanding so I could properly explain this choice, I discovered I'm either misremembering things or read something wrong at some point, because [the docs](https://go.dev/doc/modules/gomod-ref#go) clearly state it's the minimum version of Go required to compile the module. Oops.

Which, yeah, 1.18 is a better choice here. _Honestly_ I may be able to get away without bumping this at all. 🤔 After all, the code isn't _using_ generics, it's just reading them.

Let me try reverting the changes to this and go.sum, running a `go mod tidy`, and see what happens when I try to build.

Reply:
Yeah, technically 1.14 is still correct, although there's some semantic sense to 1.18. Happy to leave at 1.14.

Reply:
I was wrong about this. :( It does need to be 1.18, for the generics used in the `testdata` package.